src/com/company/Game.java

```diff
@@ -0,0 +1,186 @@
+        bowlingTeam.setupPlayers(201);
+
+    }
+    public void toss(){
+        int tossVar = (int)(Math.random()*2);
+        if(tossVar==1)      // if random number is 0 then no swap but when it is 1 then swapping the teams
+        {
+            Team t = battingTeam;
+            battingTeam = bowlingTeam;
+            bowlingTeam = t;
+        }
+        System.out.println("");
+        System.out.println("Toss is WON by: "+ battingTeam.getName()+" and they choose to bat\n");
+
+    }
+
+    //for a single ball, made a ballOutcome method which stores the particular ball outcome to players profile, prints result per ball
+    public int ballOutcome(int nextBatsman){
+        String outcome = randomOutcome();
+        if (outcome.equals("W")) {
+            battingTeam.setNotOut(battingTeam.getNotOut() - 1);
+            System.out.println(onStrike.getName() + " is OUT");
+            onStrike.setBallsFaced(onStrike.getBallsFaced() + 1);
+            bowler.setWicketsTaken(bowler.getWicketsTaken() + 1);
+
+            if(battingTeam.getNotOut()>1) {
+                onStrike = battingTeam.getMember().get(nextBatsman);
+                nextBatsman++;
+            }
+
+        }
+        else if(outcome.equals("0") || outcome.equals("2") || outcome.equals("4") || outcome.equals("6")) {
+            battingTeam.setScore(battingTeam.getScore() + Integer.parseInt(outcome));
+            onStrike.setRunsScored(onStrike.getRunsScored() + Integer.parseInt(outcome) );
+            onStrike.setBallsFaced(onStrike.getBallsFaced() + 1);
+            bowler.setRunsGiven(bowler.getRunsGiven() + Integer.parseInt(outcome));
+
+            System.out.println("Score on this ball: " + outcome + "run/s");
+        }
+        else {
+            battingTeam.setScore(battingTeam.getScore() + Integer.parseInt(outcome));
+            onStrike.setRunsScored(onStrike.getRunsScored() + Integer.parseInt(outcome) );
+            onStrike.setBallsFaced(onStrike.getBallsFaced() + 1);
+            bowler.setRunsGiven(bowler.getRunsGiven() + Integer.parseInt(outcome));
+
+            System.out.println("Score on this ball: " + outcome + "run/s");
+            Player temp =  onStrike;
+            onStrike = nonStrike;
+            nonStrike = temp;
+        }
+        bowler.setBallsDelivered(bowler.getBallsDelivered() + 1);
+        return nextBatsman;
+    }
+
+    //for an inning, made a method which is responsible for conducting overs of an inning, in it bowlers are changed, ballOutcome method is called repeatedly (also nextPlayer variable is passed in it so that which player will come to bat next can be tracked) . It is also responsible for change of strike after every over, and also prints score after every over and detailed info at the end of the inning
+    public void inning(boolean secondInning){
+        onStrike = battingTeam.getMember().get(0);
+        nonStrike = battingTeam.getMember().get(1);
+        int nextBatsman = 2,bowlAt = 0,i,j;
```

Comment:
instead of maintaining nextBatsman variable, it will always be max(onStrike.inAt(), nonStrike.inAt())

Reply:
Thanks! It is way better. Done